### PR TITLE
Registration finish ish

### PR DESCRIPTION
**Pull request by pexomat**

### CODE REVIEW

File: src/api/Events.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const fetchEvents = async () => {
+  const response = await fetch(
+    "https://sportujspolu-api.onrender.com/api/v1/events"
+  );
+
+  return await response.json();
```
Comment (GLObus303) [inline src/api/Events.ts]: I am nice to you. I dont care about formatting. But keep it consistent. Either have the new line before the return or don't, but do it for both.

CleanShot 2024-12-08 at 17 54 31

File: src/api/Events.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const fetchEvents = async () => {
```
Comment (GLObus303) [inline src/api/Events.ts]: I think we should have the api files with lowercase, they are not components nor models.

File: src/api/EventDetail.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import type { EventType } from "../model/EventType";
+
+export const fetchEventDetail = async (id: string | string[]) => {
+  const response = await fetch(
+    `https://sportujspolu-api.onrender.com/api/v1/events/${id}`
+  );
+  return response.json() as Promise<EventType>;
```
Comment (GLObus303) [inline src/api/EventDetail.ts]: The casting is completely legit in the case of window.fetch, it does not have a generic parameter allowing you to do `fetch<SomeType>`, so this is reasonable. 

But let's either have it everywhere or nowhere. We should not make each file looking like it was written by another developer/by another prompt in chatgpt :D

File: src/components/CustomInput.tsx [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
+  type: string;
   label: string;
   id: string;
   value: string;
   setValue: (value: string) => void;
 };
 
-export const TextInput: React.FC<TextInputProps> = ({
+export const CustomInput: React.FC<CustomInputProps> = ({
+  type,
```
Comment (GLObus303) [inline src/components/CustomInput.tsx]: You can default it to `text` which would save you from writing it in 90% of cases

```suggestion
  type = "text",
```

File: src/components/Login.tsx [hunk 5]
```diff
@@ -18,18 +18,20 @@ const Login: React.FC = () => {
...
-        <TextInput
+        ></CustomInput>
+        <CustomInput
+          type="password"
           label="Password"
           id="password"
           value={password}
           setValue={setPassword}
-        ></TextInput>
+        ></CustomInput>
```
Comment (GLObus303) [inline src/components/Login.tsx]: If element has no children (is empty f.e. <div></div>), you don't need to have it in pair, you can do `<div/>`.

There is usually a eslint rule for that. But let's start noticing stuff like this

File: src/components/Signup.tsx [hunk 6]
```diff
@@ -1,51 +1,67 @@
...
-    console.log("E-mail:", email);
-    console.log("Password:", password);
+
+    const requestData = {
+      email,
+      name: `${firstName} ${lastName}`,
+      password,
+    };
+
+    registerUser(requestData).then((response) => {
```
Comment (GLObus303) [inline src/components/Signup.tsx]: Handling promises with .then is completely valid. We gonna use solely async/await and there is no need to mix it up. Please do that here

**Review by GLObus303 — APPROVED** [review]

File: src/api/Events.ts (hunk 1 above)
Comment (pexomat) [reply src/api/Events.ts]: solved

File: src/components/CustomInput.tsx (hunk 4 above)
Comment (pexomat) [reply src/components/CustomInput.tsx]: true, solved

File: src/components/Login.tsx (hunk 5 above)
Comment (pexomat) [reply src/components/Login.tsx]: eslint rule implemented

File: src/api/Events.ts (hunk 2 above)
Comment (pexomat) [reply src/api/Events.ts]: resolved

File: src/components/Signup.tsx (hunk 6 above)
Comment (pexomat) [reply src/components/Signup.tsx]: undestood

File: src/api/EventDetail.ts (hunk 3 above)
Comment (pexomat) [reply src/api/EventDetail.ts]: resolved

File: src/components/Signup.tsx [hunk 7]
```diff
@@ -1,51 +1,64 @@
...
-    console.log("E-mail:", email);
-    console.log("Password:", password);
+
+    const requestData = {
+      email,
+      name: `${firstName} ${lastName}`,
+      password,
+    };
+
+    const response = await registerUser(requestData);
```
Comment (GLObus303) [inline src/components/Signup.tsx]: How about this: 
```suggestion
    const response = await registerUser({
      email,
      name: `${firstName} ${lastName}`,
      password,
    });
```

File: src/pages/event/[id].tsx [hunk 8]
```diff
@@ -12,24 +14,22 @@ const EventDetailPage: React.FC = () => {
...
-      );
-      setEvent(await response.json());
-      setIsLoading(false);
+    const loadEvent = async () => {
+      if (id) {
+        const fetchedEvent = await fetchEventDetail(id);
+        setEvent(fetchedEvent);
+        setIsLoading(false);
+      }
     };
```
Comment (GLObus303) [inline src/pages/event/[id].tsx]: Always prefer early return. 

```suggestion
    const loadEvent = async () => {
      if (!id) {
        return
      }
      
        const fetchedEvent = await fetchEventDetail(id);
        setEvent(fetchedEvent);
        setIsLoading(false);
    };
```

**Review by GLObus303 — APPROVED** [review]

File: src/components/Signup.tsx (hunk 7 above)
Comment (pexomat) [reply src/components/Signup.tsx]: resolved

File: src/pages/event/[id].tsx (hunk 8 above)
Comment (pexomat) [reply src/pages/event/[id].tsx]: resolved

**Pull request merged**